pkg/policy/distillery_test.go

```diff
@@ -1127,3 +1160,167 @@ func Test_AllowAll(t *testing.T) {
+	mapEntryL3L4__WorldDeny = MapStateEntry{
+		ProxyPort:        0,
+		DerivedFromRules: labels.LabelArrayList{nil},
+		IsDeny:           true,
+		owners:           map[MapStateOwner]struct{}{},
+	}
+	mapEntryL3L4__WorldDenyWithLabels = MapStateEntry{
+		ProxyPort:        0,
+		DerivedFromRules: worldLabelArrayList,
+		IsDeny:           true,
+		owners:           map[MapStateOwner]struct{}{},
+	}
+
+	worldIPIdentity        = identity.NumericIdentity(16324)
+	worldCIDR              = api.CIDR("192.0.2.3/32")
+	lblWorldIP             = labels.ParseSelectLabelArray(fmt.Sprintf("%s:%s", labels.LabelSourceCIDR, worldCIDR))
+	ruleL3L4__AllowWorldIP = api.NewRule().
+				WithIngressRules([]api.IngressRule{{
+			IngressCommonRule: api.IngressCommonRule{
+				FromCIDR: api.CIDRSlice{worldCIDR},
+			},
+		}}).
+		WithEgressRules([]api.EgressRule{{
+			EgressCommonRule: api.EgressCommonRule{
+				ToCIDR: api.CIDRSlice{worldCIDR},
+			},
+		}}).
+		WithEndpointSelector(api.WildcardEndpointSelector)
+
+	worldSubnetIdentity = identity.NumericIdentity(16325)
+	worldSubnet         = api.CIDR("192.0.2.0/24")
+	worldSubnetRule     = api.CIDRRule{
+		Cidr: worldSubnet,
+	}
+	lblWorldSubnet       = cidr.GetCIDRLabels(netip.MustParsePrefix(string(worldSubnet)))
+	ruleL3L4__DenySubnet = api.NewRule().
+				WithIngressDenyRules([]api.IngressDenyRule{{
+			IngressCommonRule: api.IngressCommonRule{
+				FromCIDRSet: api.CIDRRuleSlice{worldSubnetRule},
+			},
+		}}).
+		WithEgressDenyRules([]api.EgressDenyRule{{
+			EgressCommonRule: api.EgressCommonRule{
+				ToCIDRSet: api.CIDRRuleSlice{worldSubnetRule},
+			},
+		}}).
+		WithEndpointSelector(api.WildcardEndpointSelector)
+	mapKeyL3L4__SubnetIngress = Key{worldSubnetIdentity.Uint32(), 0, 0, trafficdirection.Ingress.Uint8()}
+	mapKeyL3L4__SubnetEgress  = Key{worldSubnetIdentity.Uint32(), 0, 0, trafficdirection.Egress.Uint8()}
+
+	ruleL3L4__DenySmallerSubnet = api.NewRule().
+					WithIngressDenyRules([]api.IngressDenyRule{{
+			IngressCommonRule: api.IngressCommonRule{
+				FromCIDRSet: api.CIDRRuleSlice{api.CIDRRule{Cidr: worldCIDR}},
+			},
+		}}).
+		WithEgressDenyRules([]api.EgressDenyRule{{
+			EgressCommonRule: api.EgressCommonRule{
+				ToCIDRSet: api.CIDRRuleSlice{api.CIDRRule{Cidr: worldCIDR}},
+			},
+		}}).
+		WithEndpointSelector(api.WildcardEndpointSelector)
+	ruleL3L4__AllowLargerSubnet = api.NewRule().
+					WithIngressRules([]api.IngressRule{{
+			IngressCommonRule: api.IngressCommonRule{
+				FromCIDRSet: api.CIDRRuleSlice{api.CIDRRule{Cidr: worldSubnet}},
+			},
+		}}).
+		WithEgressRules([]api.EgressRule{{
+			EgressCommonRule: api.EgressCommonRule{
+				ToCIDRSet: api.CIDRRuleSlice{api.CIDRRule{Cidr: worldSubnet}},
+			},
+		}}).
+		WithEndpointSelector(api.WildcardEndpointSelector)
+
+	mapKeyL3L4__SmallerSubnetIngress = Key{worldIPIdentity.Uint32(), 0, 0, trafficdirection.Ingress.Uint8()}
+	mapKeyL3L4__SmallerSubnetEgress  = Key{worldIPIdentity.Uint32(), 0, 0, trafficdirection.Egress.Uint8()}
```

Comment:
It looks like these are only evaluating the L3 conflict cases. It would be good to have corner cases to test the subsets in case of mixed L3 and L3L4 rules, eg a deny for a smaller L3 that conflicts with an allow for a wider L3, or intersection between an L3-only and an L4-only rule, one deny and one not.